lib/Dialect/FIRRTL/Transforms/InferResets.cpp

```diff
@@ -1142,8 +1166,16 @@ static FIRRTLBaseType updateType(FIRRTLBaseType oldType, unsigned fieldID,
 /// Update the reset type of a specific field.
 bool InferResetsPass::updateReset(FieldRef field, FIRRTLBaseType resetType) {
   // Compute the updated type.
-  auto oldType = field.getValue().getType().cast<FIRRTLBaseType>();
-  auto newType = updateType(oldType, field.getFieldID(), resetType);
+  auto oldType = field.getValue().getType().cast<FIRRTLType>();
+  FIRRTLType newType =
+      TypeSwitch<FIRRTLType, FIRRTLType>(oldType)
+          .Case<FIRRTLBaseType>([&](auto base) {
+            return updateType(base, field.getFieldID(), resetType);
+          })
+          .Case<RefType>([&](auto ref) {
+            return RefType::get(
+                updateType(ref.getType(), field.getFieldID(), resetType));
+          });
```

Comment:
This type of stuff feels a lot like we want a `map` function defined for `FIRRTLType`.

Reply:
Good call, agreed! Will add something like this in a follow-up PR (it'll be useful in a few places..).